### PR TITLE
Add UCSF base docker image

### PR DESCRIPTION
**Pull request by jrjohnson**
Creates a base `Dockerfile` which will ship our code and plugins as well as a a `docker-composer.yml` file which we can use in development.

Work in progress todo:
- [ ] Don't use the moodlehq image, it isn't suitable for production
- [ ] Add configuration files so we don't have to do the setup process each time
- [ ] Figure out what goes where (is it secure to put the whole thing in `/var/www/html`?

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      MOODLE_DOCKER_DBPASS: "m@0dl3ing"
+    ports:
+            - "8000:80"
+    volumes:
+        - ./:/var/www/html:delegated
+  db:
+    image: mariadb:10.4
+    environment:
+      MYSQL_USER: moodle
+      MYSQL_PASSWORD: "m@0dl3ing"
```
Comment (Trott) [inline docker-compose.yml]: Uh......

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+version: "2"
+services:
+  webserver:
+    build: ./
+    environment:
+      MOODLE_DOCKER_DBTYPE: mariadb
+      MOODLE_DOCKER_DBNAME: moodle
+      MOODLE_DOCKER_DBUSER: moodle
+      MOODLE_DOCKER_DBPASS: "m@0dl3ing"
```
Comment (Trott) [inline docker-compose.yml]: Also Uh.....

Comment (jrjohnson) [reply docker-compose.yml]: I stole it from https://github.com/moodlehq/moodle-docker/blob/master/base.yml#L14 also public, and also hopefully not used by anyone for real stuff.

**Comment by jrjohnson** [comment]
@ctam has gone well beyond this now in `UCSFCLE_35_DEV` which supersedes this.

**Pull request closed without merging**